### PR TITLE
Update the `Truncate` component to accept `:strong` as a tag.

### PR DESCRIPTION
**Pull request by Artemisia-Absinthium**
# Changes
Added the `:strong` tag in the `Truncate` component tag options.
Added a test for the `:strong` tag and one other for the `:p` tag.

# Why
This change follows an internal slack discussion about adding strong as a tag option. (See issue https://github.com/primer/view_components/issues/641)

# Linked issues
Close https://github.com/primer/view_components/issues/641

### CODE REVIEW

File: test/components/truncate_test.rb [hunk 1]
```diff
@@ -17,9 +17,22 @@ def test_renders_truncated_span
...
+    render_inline(Primer::Truncate.new(tag: :p)) { "content" }
+
+    assert_selector("p.css-truncate.css-truncate-overflow", text: "content")
+  end
+
+  def test_renders_truncated_strong
+    render_inline(Primer::Truncate.new(tag: :strong)) { "content" }
+
+    assert_selector("strong.css-truncate.css-truncate-overflow", text: "content")
+  end
```
Comment (manuelpuyol) [inline test/components/truncate_test.rb:30]: what do you think of having a more generic test, so we don't have to keep adding a test every time we change a tag?

```rb
def test_renders_multiple_tags
  Primer::Truncate::TAG_OPTIONS.each do |tag|
    render_inline(Primer::Truncate.new(tag: tag)) { "content" }

    assert_selector("#{tag}.css-truncate.css-truncate-overflow", text: "content")
  end
end
```

Comment (Artemisia-Absinthium) [reply test/components/truncate_test.rb:30]: Yes! I like it!

Still learning a lot with Ruby. I will change to that one :)

File: app/components/primer/truncate.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class Truncate < Primer::Component
     status :beta
 
     DEFAULT_TAG = :div
-    TAG_OPTIONS = [DEFAULT_TAG, :span, :p].freeze
+    TAG_OPTIONS = [DEFAULT_TAG, :span, :p, :strong].freeze
```
Comment (khiga8) [inline app/components/primer/truncate.rb:9]: While we're here, do you think we should consider other similar text elements like, `<b>` and `<i>`?

Comment (Artemisia-Absinthium) [reply app/components/primer/truncate.rb:9]: I was looking up the [phrasing content tags](https://html.spec.whatwg.org/#phrasing-content) and I feel there is a lot to add if we want to be exhaustive.
`<b>` and `<i>` for sure!

I feel `<em>`, `<samp>`, `<small>` and `<u>` can also be used with truncated text.

And `<bdi>` and `<bdo>` for bidirecitonal text?

I don't see anything bad about giving more semantic options to the user, but I don't know how we want to tackle that in the project in general?

Comment (khiga8) [reply app/components/primer/truncate.rb:9]: Good point. Let's leave it out for now :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -36,6 +36,10 @@
 
     *Kate Higa*
 
+* Update the `Truncate` component to accept `:strong` as a tag.
+
+    *Amélia Chavot*
+
```
Comment (manuelpuyol) [inline CHANGELOG.md:42]: I think this could be in the `Updates` section! :)

**Review by manuelpuyol — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (Artemisia-Absinthium) [reply CHANGELOG.md:42]: To the update section then!

Do we have some rules for where changes should be written down? Or do we do it by intuition like I did? ^^'

Comment (joelhawksley) [reply CHANGELOG.md]: @Artemisia-Absinthium good question! Would you be up for updating our contributing guidelines to mention these sections?

Comment (Artemisia-Absinthium) [reply CHANGELOG.md]: Yep! I will draft a new PR for that. :)

**Review by joelhawksley — APPROVED** [review]
❤️ 🧡 💛 💚 💙 💜

**Pull request merged**